curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64caf1be15606d5814c3387b.md

```diff
@@ -0,0 +1,357 @@
+---
+id: 64caf1be15606d5814c3387b
+title: Step 86
+challengeType: 0
+dashedName: step-86
+---
+
+# --description--
+
+Add another conditional statement that checks if the player's `y` position plus the player's height is greater than or equal to the platform's `y` position.
+
+Below that, add another conditional statement that checks if the player's `y` position is less than or equal to the sum of the platform's `y` position plus the platform's height.
+
+# --hints--
+
+TODO: add tests. Tried to add tests earlier but they were not working.
+
+```js
+
+```
```

Comment:
These work:

You should not alter the existing `platformDetectionRules` array and its content.

```js
assert.match(code, /const\s*platformDetectionRules\s+=\s*\[\s*player\.position\.x\s*>=\s*platform\.position\.x\s*-\s*player\.width\s*\/\s*2,?\s*player\.position\.x\s*<=\s*platform\.position\.x\s*\+\s*platform\.width\s*-\s*player\.width\s*\/\s*3,\s*.*\s*.*\s*\];?/)
```

You should have a conditional statement that checks if the player's `y` position plus the player's height is greater than or equal to the platform's `y` position

```js
assert.match(code, /const\s*platformDetectionRules\s+=\s*\[\s*player\.position\.x\s*>=\s*platform\.position\.x\s*-\s*player\.width\s*\/\s*2,?\s*player\.position\.x\s*<=\s*platform\.position\.x\s*\+\s*platform\.width\s*-\s*player\.width\s*\/\s*3,\s*player\.position\.y\s*\+\s*player\.height\s*>=\s*platform\.position\.y,\s*.*\s*\];?/)
```

You should have a conditional statement that checks if the player's `y` position is less than or equal to the sum of the platform's `y` position plus the platform's height.

```js
assert.match(code, /const\s*platformDetectionRules\s+=\s*\[\s*player\.position\.x\s*>=\s*platform\.position\.x\s*-\s*player\.width\s*\/\s*2,?\s*player\.position\.x\s*<=\s*platform\.position\.x\s*\+\s*platform\.width\s*-\s*player\.width\s*\/\s*3,\s*player\.position\.y\s*\+\s*player\.height\s*>=\s*platform\.position\.y,\s*player\.position\.y\s*<=\s*platform\.position\.y\s*\+\s*platform\.height,?\s*\];?/)
```